From 0c0cc1f62bc5efc8eeda299d656dd3d99bb3e4d5 Mon Sep 17 00:00:00 2001
From: 097115 <097115@gmail.com>
Date: Wed, 19 Jan 2022 22:01:31 +0200
Subject: [PATCH] perf: Makefile: $(prefix) updated, clipboard tools updated,
 renamed to 'scim'; perf: help.c, color.c: change colors of help page (febe708
 reversed)

---
 src/Makefile | 20 ++++++++++----------
 src/color.c  |  1 -
 src/help.c   |  2 +-
 3 files changed, 11 insertions(+), 12 deletions(-)

diff --git a/src/Makefile b/src/Makefile
index defd77a..f075b17 100755
--- a/src/Makefile
+++ b/src/Makefile
@@ -1,8 +1,8 @@
 # Specify the name of the resulting executable file
-name = sc-im
+name = scim
 
 # The base directory where everything should be installed.
-prefix  = /usr/local
+prefix  = ${HOME}/usr/local/bundled/scim
 
 EXDIR   = $(prefix)/bin
 HELPDIR = $(prefix)/share/$(name)
@@ -13,9 +13,9 @@ MANDIR  = $(prefix)/share/man/man1
 
 # History directory, relative to $HOME
 HISTORY_DIR= .cache
-HISTORY_FILE=sc-iminfo
+HISTORY_FILE=sciminfo
 # Configuration directory, relative to $HOME
-CONFIG_DIR= .config/sc-im
+CONFIG_DIR= .config/scim
 CONFIG_FILE=scimrc
 
 # Change these to your liking or use `make CC=gcc` etc
@@ -43,7 +43,7 @@ CFLAGS += -DHISTORY_FILE=\"$(HISTORY_FILE)\" -DHISTORY_DIR=\"$(HISTORY_DIR)\"
 # Configuration file, relative to CONFIG_DIR directory
 CFLAGS += -DCONFIG_FILE=\"$(CONFIG_FILE)\" -DCONFIG_DIR=\"$(CONFIG_DIR)\"
 # Input mode history. Same as previous, but for insert mode commands
-CFLAGS += -DINS_HISTORY_FILE=\"sc-iminfo\"
+CFLAGS += -DINS_HISTORY_FILE=\"sciminfo\"
 # Comment out to disable undo/redo support
 CFLAGS += -DUNDO
 # Maximum number of rows in spreadsheet. Up to 1048576
@@ -60,14 +60,14 @@ CFLAGS += -DMOUSE
 #to copy to tmux clipboard:
 #CFLAGS += -DDEFAULT_COPY_TO_CLIPBOARD_CMD=\""tmux load-buffer"\"
 #to copy to X clipboard:
-CFLAGS += -DDEFAULT_COPY_TO_CLIPBOARD_CMD=\""xclip -i -selection clipboard <"\"
+CFLAGS += -DDEFAULT_COPY_TO_CLIPBOARD_CMD=\""${XCOPY}"\"
 #to copy to OSX clipboard:
 #CFLAGS += -DDEFAULT_COPY_TO_CLIPBOARD_CMD=\""pbcopy <"\"
 #
 # Choose one of the proposed commands for pasting from different clipboards:
 # You can later change it at runtime.
 #CFLAGS += -DDEFAULT_PASTE_FROM_CLIPBOARD_CMD=\""tmux show-buffer"\"
-CFLAGS += -DDEFAULT_PASTE_FROM_CLIPBOARD_CMD=\""xclip -o -selection clipboard"\"
+CFLAGS += -DDEFAULT_PASTE_FROM_CLIPBOARD_CMD=\""${XPASTE}"\"
 #CFLAGS += -DDEFAULT_PASTE_FROM_CLIPBOARD_CMD=\""pbpaste"\"
 
 # Command to open file or link under cursor
@@ -182,7 +182,7 @@ install :
 	install -m 644 doc $(DESTDIR)$(HELPDIR)/$(name)_help
 	install -m 644 $(wildcard plot_*) $(DESTDIR)$(HELPDIR)/
 	install -d $(DESTDIR)$(MANDIR)/
-	install -m 644 sc-im.1 $(DESTDIR)$(MANDIR)/$(name).1
+	install -m 644 scim.1 $(DESTDIR)$(MANDIR)/$(name).1
 
 uninstall :
 	-rm $(DESTDIR)$(prefix)/bin/$(name)
@@ -223,9 +223,9 @@ man_install:
 	@cp -r ../docs/man/man3/ /usr/local/share/man/
 	mandb
 
-# "sc-im" MUST match what is in Doxyfile `MAN_EXTENSION = .sc-im.3`
+# "scim" MUST match what is in Doxyfile `MAN_EXTENSION = .scim.3`
 man_uninstall:
-	@rm -rf /usr/local/share/man/man3/*sc-im.3
+	@rm -rf /usr/local/share/man/man3/*scim.3
 	@mandb
 
 clean:
diff --git a/src/color.c b/src/color.c
index c8aeedb..4df7610 100644
--- a/src/color.c
+++ b/src/color.c
@@ -144,7 +144,6 @@ void start_default_ucolors() {
     ucolors[ CELL_CONTENT    ].bold = 1;
     ucolors[ INPUT           ].fg = WHITE;
     ucolors[ INPUT           ].bg = DEFAULT_COLOR;
-    ucolors[ NORMAL          ].bold = 1;
     ucolors[ NORMAL          ].fg = WHITE;
     ucolors[ NORMAL          ].bg = DEFAULT_COLOR;
     ucolors[ CELL_ERROR      ].fg = RED;
diff --git a/src/help.c b/src/help.c
index 922ff34..956c43a 100644
--- a/src/help.c
+++ b/src/help.c
@@ -165,7 +165,7 @@ void help() {
 
     wbkgd(main_win, COLOR_PAIR((ucolors[DEFAULT].fg+1) * (COLORS) + ucolors[DEFAULT].bg + 2));
     ui_set_ucolor(main_win, &ucolors[NORMAL], DEFAULT_COLOR);
-    ui_set_ucolor(input_win, &ucolors[NORMAL], DEFAULT_COLOR);
+//     ui_set_ucolor(input_win, &ucolors[NORMAL], DEFAULT_COLOR);
     wtimeout(input_win, -1);
     noecho();
     curs_set(0);
-- 
2.7.4 (Apple Git-66)

